blockchain/chain.go

```diff
@@ -766,20 +764,29 @@ func (b *BlockChain) calcSequenceLock(tx *btcutil.Tx, utxoView *UtxoViewpoint,
 	// activated.
 	sequenceLock := &SequenceLock{Seconds: -1, BlockHeight: -1}
 
+	// Obtain the latest BIP9 version bits state for the CSV-package
+	// soft-fork deployment. The adherence of sequence locks depends on the
+	// current soft-fork state.
+	csvState, err := b.deploymentState(node, chaincfg.DeploymentCSV)
```

Comment:
Nice, I like the suggestion. This'll also indirectly speed up mempool acceptance/processing since it doesn't need to _unconditionally_ query the the version bit state.